internal/sidekick/internal/parser/discovery/methods_test.go

```diff
@@ -15,19 +15,32 @@
 package discovery
 
 import (
+	"slices"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/googleapis/librarian/internal/sidekick/internal/api"
+	"github.com/googleapis/librarian/internal/sidekick/internal/api/apitest"
 )
 
 func TestMakeServiceMethods(t *testing.T) {
 	model, err := ComputeDisco(t, nil)
 	if err != nil {
 		t.Fatal(err)
 	}
-
 	id := "..zones.get"
+	// The `model.Messages` is is automatically generated. We want the synthetic
+	// request messages to require ad-hoc generation. Each codec may need to
+	// do something special to avoid clashes.
+	index := slices.IndexFunc(model.Messages, func(a *api.Message) bool { return a.ID == id })
+	if index != -1 {
+		t.Errorf("synthetic request messages should not be in the `model.Messages` list")
```

Comment:
I feel like they should be in the list, they should just be tagged as synthetic. :shrug: